spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/ReactivePulsarConsumerFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core.reactive;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageConsumer;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageConsumerSpec;
+
+/**
+ * Pulsar reactive consumer factory interface.
+ *
+ * @param <T> payload type for the consumer.
+ * @author Christophe Bornet
+ */
+public interface ReactivePulsarConsumerFactory<T> {
+
+	ReactiveMessageConsumer<T> createConsumer(Schema<T> schema, ReactiveMessageConsumerSpec spec);
```

Comment:
Should we also add `ReactiveMessageConsumer<T> createConsumer(Schema<T> schema)` ?

Reply:
I noticed this differs from the RPSF in that the producer variant does not allow user to pass in a spec but rather allows them to get the spec.

Reply:
Yes we can do it also for RSPF. RSPF is consistent with PulsarProducerFactory (ie PulsarProducerFactory doesn't have properties in its methods whereas PulsarConsumerFactory). I think we should add `properties` in the  PulsarProducerFactory methods.

Reply:
Ahh ok. I see it now. 

>  I think we should add properties in the PulsarProducerFactory methods.

In the PPF there is already somewhat of a method signature explosion - adding `properties` would add to that. 

I would love for produce/consume be consistent in this area - one way or the other. Because of 1) the signature explosion and 2) changing the PPF would be quite a ripple I would prefer if we make the consumers follow the pattern of the producers. Then the global pattern would be:
- `<init>` gets the default spec/props
- expose the default spec/props via `get<Producer|Consumer><Spec|Props>`

Wdyt?

Reply:
So you can't customize props per Consumer ?

Reply:
In the PPF the hook to customize is via the ProducerBuilderCustomizer. Should we do the same here?

Reply:
Oh yes. In that case, we do have a `ReactiveMessageSenderBuilderCustomizer`.

Reply:
So I'll add the `getReactiveMessageSenderSpec ` and `createConsumer(Schema<T> schema)`.
We should probably also have `createConsumer(Schema<T> schema)` for the imperative `PulsarConsumerFactory`.

Reply:
Done.
Also added ReactiveMessageSenderBuilderCustomizer for consistency (I'll do a PR to align the imperative API)

Reply:
Using a customizer is much better. It allows to set `maxInFlight`, the `cache` or `maxConcurrentSenderSubscriptions`